### PR TITLE
Remove poetry from CI

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes

This PR switches to `pip` for installing ZenML in development environments and the CI.
`poetry` is removed everywhere except the release publishing.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -117,20 +117,18 @@ the ["fork-and-pull" Git workflow](https://github.com/susam/gitpr)
...
 ### 🧐 Linting, formatting, and tests
 
-ZenML is mainly developed using [Poetry](https://python-poetry.org/) as the
-dependency management system. In order to
-install all core dev-dependencies, do:
+To install ZenML from your local checked out files includin all core dev-dependencies, run:
 
 ```
-poetry install
+pip install -e ".[dev]"
```
Comment (strickvl) [inline CONTRIBUTING.md:123]: Wouldn't they want the server as well for developing with ZenML?

File: pyproject.toml [hunk 2]
```diff
@@ -85,55 +85,50 @@ jinja2-time = { version = "^0.2.0", optional = true }
...
+types-requests = { version = "^2.27.11", optional = true }
+types-setuptools = { version = "^57.4.2", optional = true }
+types-six = { version = "^1.16.2", optional = true }
+types-termcolor = { version = "^1.1.2", optional = true }
+types-psutil = { version = "^5.8.13", optional = true }
 # test framework dependencies
-docker-compose = "^1.28"
+docker-compose = { version = "^1.28", optional = true }
+
+[tool.poetry.extras]
```
Comment (strickvl) [inline pyproject.toml:128]: why is this still referring to Poetry?

File: src/zenml/README.md [hunk 3]
```diff
@@ -18,43 +15,41 @@ git clone git@github.com:zenml-io/zenml.git
...
-poetry shell
+pip install --upgrade pip
 ```
 
 ## Install dependencies
+
 Then from the root of the package:
 ```bash
-poetry install
+pip install -e ".[dev]"
```
Comment (strickvl) [inline src/zenml/README.md:40]: wouldn't they also want the server dependencies?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -16,23 +16,23 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  poetry-install:
+  install:
```
Comment (strickvl) [inline .github/workflows/ci.yml:19]: `install` sounds a bit close to a reserved word; perhaps worth making less generic just in case?

**Review by strickvl — COMMENTED** [review]
This looks good to me and assume it has been tested alongside the CI / local installs etc. Some questions only...

File: CONTRIBUTING.md (hunk 1 above)
Comment (schustmi) [reply CONTRIBUTING.md:123]: Potentially but not necessarily? I just copied the previous behaviour tbh, which also didn't install the server extra. I'll change it

Comment (strickvl) [reply CONTRIBUTING.md:123]: Yeah either add both by default, or explain the different combinations they might want to use depending on what they're doing?

File: pyproject.toml (hunk 2 above)
Comment (schustmi) [reply pyproject.toml:128]: This is for the publishing to pypi which still uses poetry. However this poetry value is a synonym that also works for pip (See https://peps.python.org/pep-0621/)

File: .github/workflows/ci.yml (hunk 4 above)
Comment (schustmi) [reply .github/workflows/ci.yml:19]: Mhm it seems to work without any issues, what other name would you suggest?

Comment (strickvl) [reply .github/workflows/ci.yml:19]: install-environment? install-python-environment?

**Review by stefannica — APPROVED** [review]
YES !

**Review by strickvl — APPROVED** [review]

**Review by AlexejPenner — APPROVED** [review]
:seal:

**Pull request merged**